### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by lukasmerten**
Updates the documentation.

1) Should fix the the problems reported in issue #346. At least locally on my machine this works.
2) Adds some additional explanation, fixes some bugs and adjusts formatting.

### CODE REVIEW

File: doc/pages/galactic_cosmic_rays.rst [hunk 1]
```diff
@@ -1,9 +1,36 @@
...
+Advection in non-divergence-free velocity fields causes adiabatic energy
+changes, which can be modeled following the next examples.
+
+.. toctree::
+   example_notebooks/Diffusion/AdiabaticCooling.ipynb
+
+Gas Densities
+^^^^^^^^^^^^^
+
+The last to examples show how to use the gas density fields. They can be
```
Comment (avvliet) [inline doc/pages/galactic_cosmic_rays.rst:29]: Typo: to -> two

File: doc/pages/example_notebooks/Diffusion/AdiabaticCooling.ipynb [hunk 2]
```diff
@@ -27,13 +27,15 @@
...
-    "%matplotlib notebook\n",
+    "%matplotlib inline\n",
     "\n",
     "import pandas as pd\n",
     "import matplotlib.pyplot as plt\n",
     "import seaborn as sns\n",
     "import numpy as np\n",
     "from scipy.optimize import curve_fit\n",
+    "import sys\n",
+    "sys.path.append('../../../../build/')\n",
```
Comment (avvliet) [inline doc/pages/example_notebooks/Diffusion/AdiabaticCooling.ipynb:38]: Is this necessary?

File: doc/pages/example_notebooks/Diffusion/DiffusionValidationI.v4.ipynb [hunk 3]
```diff
@@ -27,14 +27,16 @@
...
+    "%matplotlib inline\n",
     "\n",
     "import pandas as pd\n",
     "import matplotlib.pyplot as plt\n",
     "import seaborn as sns\n",
     "import numpy as np\n",
     "from scipy.stats import chisquare\n",
     "from scipy.integrate import quad\n",
+    "import sys\n",
+    "sys.path.append('../../../../build/')\n",
```
Comment (avvliet) [inline doc/pages/example_notebooks/Diffusion/DiffusionValidationI.v4.ipynb:39]: Here again, why is this path appended? Seems like bad practice for a general example.

File: doc/pages/example_notebooks/Diffusion/DiffusionValidationII.v4.ipynb [hunk 4]
```diff
@@ -36,7 +36,8 @@
     "from scipy.stats import chisquare\n",
     "from scipy.integrate import quad\n",
     "from scipy.stats import anderson\n",
-    "\n",
+    "import sys\n",
+    "sys.path.append('../../../../build/')\n",
```
Comment (avvliet) [inline doc/pages/example_notebooks/Diffusion/DiffusionValidationII.v4.ipynb:40]: Same here

File: doc/pages/galactic_cosmic_rays.rst (hunk 1 above)
Comment (lukasmerten) [reply doc/pages/galactic_cosmic_rays.rst]: Fixed.

File: doc/pages/example_notebooks/Diffusion/AdiabaticCooling.ipynb (hunk 2 above)
Comment (lukasmerten) [reply doc/pages/example_notebooks/Diffusion/AdiabaticCooling.ipynb:38]: No, this is not necessary and probably also bad practice. I run the examples locally in a wrong environment and forgot to remove the import before pushing.

File: doc/pages/example_notebooks/Diffusion/DiffusionValidationI.v4.ipynb (hunk 3 above)
Comment (lukasmerten) [reply doc/pages/example_notebooks/Diffusion/DiffusionValidationI.v4.ipynb]: Fixed.

File: doc/pages/example_notebooks/Diffusion/DiffusionValidationII.v4.ipynb (hunk 4 above)
Comment (lukasmerten) [reply doc/pages/example_notebooks/Diffusion/DiffusionValidationII.v4.ipynb]: Fixed.

**Review by avvliet — APPROVED** [review]
It all looks good to me, and I think it's important to fix this as soon as possible.

**Pull request merged**